mne/annotations.py

```diff
@@ -236,23 +248,31 @@ def append(self, onset, duration, description):
 
         Parameters
         ----------
-        onset : float
+        onset : float | array-like
             Annotation time onset from the beginning of the recording in
             seconds.
-        duration : float
+        duration : float | array-like
             Duration of the annotation in seconds.
-        description : str
+        description : str | array-like
             Description for the annotation. To reject epochs, use description
             starting with keyword 'bad'
 
         Returns
         -------
         self : mne.Annotations
             The modified Annotations object.
-        """
+
+        Notes
+        -----
+        The array-like support for arguments allows this to be used similarly
+        to not only ``list.append``, but also
+        `list.extend <https://docs.python.org/3/library/stdtypes.html#mutable-sequence-types>`__.
+        """  # noqa: E501
+        onset, duration, description = _check_ods(onset, duration, description)
         self.onset = np.append(self.onset, onset)
         self.duration = np.append(self.duration, duration)
         self.description = np.append(self.description, description)
+        self._sort()
```

Comment:
The problem of doing it like this is that we lose the fact that self was already sorted. Do we care?

Reply:
based on: 
https://stackoverflow.com/questions/27916710/numpy-merge-sorted-array-to-an-new-array
https://stackoverflow.com/questions/35932101/numpy-append-vs-concatenate

I guess we cannot do better. (That's good to know)

Reply:
acutally `_sort` is not not using the sort from `numpy`, then my question remain. do we care?

Reply:
This is a performance question, let's wait to see if we need better perf (I doubt it, especially with `extend`-like support as a workaround)